target_tools/client_info.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""Helper functions for deriving various info from client"""
+import re
+
+from target_python_sdk.utils import is_string
+from target_python_sdk.utils import parse_int
+
+BROWSER_MATCHERS_LIST = [
```

Comment:
While this implementation is fine, I want to point out that it was written as a lean way to parse user-agent strings rather than include a heavier-weight dependency in at.js builds.   It does a "good enough" job, although it may not cover every single user agent known to man.  So for server-side SDKs like python, you can get better support of parsing this string by relying on a package that includes more robust detection and device definitions.  And on the server, there is no size drawback to relying on a package.  I recommend switching to use a popular open-source package instead for user-agent string parsing. I did a quick google search and found this one, for example, https://pypi.org/project/httpagentparser/